libs/server/Lua/LuaStateWrapper.cs

```diff
@@ -500,6 +511,36 @@ internal void PushConstantString(int constStringRegistryIndex)
 
         // Rarely used
 
+        /// <summary>
+        /// This should be used to set all debug hooks for Lua when we know no other thread will be executing.
+        /// </summary>
+        internal unsafe void SetHook(delegate* unmanaged[Cdecl]<nint, nint, void> hook, LuaHookMask mask, int count)
+        {
+            NativeMethods.SetHook(state, hook, mask, count);
+        }
+
+        /// <summary>
+        /// This should be used to set all debug hooks for Lua when multiple threads are involved.
+        /// 
+        /// This can fail if there's a thread race to close the state.
+        /// 
+        /// This is the ONLY thread-safe method on <see cref="LuaStateWrapper"/>.
+        /// </summary>
+        internal unsafe bool TrySetHook(delegate* unmanaged[Cdecl]<nint, nint, void> hook, LuaHookMask mask, int count)
+        {
+            lock (hookLock)
+            {
+                if (state == 0)
+                {
+                    return false;
+                }
+
+                NativeMethods.SetHook(state, hook, mask, count);
+            }
+
+            return true;
+        }
+
```

Comment:
Check warning on line 676 in libs/server/Lua/LuaStateWrapper.cs: XML comment has cref attribute 'SetAllocFunction' that could not be resolved

Reply:
There are also 3-4 other warnings in the PR, please check those.

Reply:
Believe I've hit these all.

Perhaps we should WAE these for the future?

Reply:
Good point. Tal is looking to making this change.